### PR TITLE
Addressing PR comments.

### PR DESCRIPTION
**Pull request by jwarmkessel**
Correclty re-implemented NSDictionary category to transform dictionary into a string. Refactored displacement uploader. Passing in HealthStore as a parameter into Healthkit Collector. Refactored file security key so user can specify the level of protection.

### CODE REVIEW

File: APCAppCore/APCAppCore/Library/Categories/NewDataCollector/APCPassiveDisplacementTrackingDataUploader.m [hunk 1]
```diff
@@ -176,114 +153,77 @@ - (void)didReceiveUpdateWithLocationManager:(CLLocationManager*)manager withUpda
...
-{
-    if (!_baseTrackingLocation)
+    if (self.folder)
     {
-        if (self.folder)
+        if ([[NSFileManager defaultManager] fileExistsAtPath:filePath])
         {
-            if ([[NSFileManager defaultManager] fileExistsAtPath:[self baseTrackingFilePath]])
+            NSError*    error       = nil;
+            NSString*   jsonString  = [NSString stringWithContentsOfFile:filePath
```
Comment (peculiar) [inline APCAppCore/APCAppCore/Library/Categories/NewDataCollector/APCPassiveDisplacementTrackingDataUploader.m]: "jsonString" is not a descriptive name.

File: APCAppCore/APCAppCore/Library/Categories/NewDataCollector/APCPassiveDisplacementTrackingDataUploader.m [hunk 2]
```diff
@@ -176,114 +153,77 @@ - (void)didReceiveUpdateWithLocationManager:(CLLocationManager*)manager withUpda
...
-                                                                       error:&error];
-                
-                if (!jsonString)
+                if (error)
                 {
-                    if (error)
-                    {
-                        APCLogError2(error);
-                    }
+                    APCLogError2(error);
```
Comment (peculiar) [inline APCAppCore/APCAppCore/Library/Categories/NewDataCollector/APCPassiveDisplacementTrackingDataUploader.m]: If an error occurs, should it be communicated back to the caller?

File: APCAppCore/APCAppCore/Library/Categories/NewDataCollector/Data Sink/APCPassiveDataSink.m [hunk 3]
```diff
@@ -299,6 +304,8 @@ - (void)uploadWithDataArchiverAndUploader
             APCLogError2(error);
```
Comment (peculiar) [inline APCAppCore/APCAppCore/Library/Categories/NewDataCollector/Data Sink/APCPassiveDataSink.m]: Should the error be communicated by to the caller?

File: APCAppCore/APCAppCore/Library/Categories/NewDataCollector/APCPassiveDisplacementTrackingDataUploader.m (hunk 2 above)
Comment (jwarmkessel) [inline APCAppCore/APCAppCore/Library/Categories/NewDataCollector/APCPassiveDisplacementTrackingDataUploader.m]: No, this isn't necessary.

File: APCAppCore/APCAppCore/Library/Categories/NewDataCollector/Data Sink/APCPassiveDataSink.m (hunk 3 above)
Comment (jwarmkessel) [inline APCAppCore/APCAppCore/Library/Categories/NewDataCollector/Data Sink/APCPassiveDataSink.m]: No, this isn't necessary.

**Pull request merged**